plugins/user-authenticators/ldap/src/main/java/org/apache/cloudstack/ldap/LdapManagerImpl.java

```diff
@@ -125,10 +134,28 @@ public void onPublishMessage(String senderAddress, String subject, Object args)
                 }
             }
         });
-
-        return true;
     }
 
+    private void addDomainRemovalListener() {
+        messageBus.subscribe(DomainManager.MESSAGE_REMOVE_DOMAIN_EVENT, new MessageSubscriber() {
+            @Override
+            public void onPublishMessage(String senderAddress, String subject, Object args) {
+                try {
+                    final Domain domain = domainDao.findByIdIncludingRemoved((Long) args);
+                    long domainId = domain.getId();
+                    LdapTrustMapVO ldapTrustMapVO = _ldapTrustMapDao.findByDomainId(domainId);
```

Comment:
I think the method should be similar as `listByDomainId`, it should return a list of VOs.

Reply:
there is no consitency in these names, sorry. This one returns a single `LdapTrustMapVO`.

Reply:
@DaanHoogland 
do you mean the record is for a domain, not  accounts ?

Reply:
good point, I think there is but I will use `searchByDomainId()` anyway and iterate over the results to be sure.